instrumentation/spring/spring-integration-4.1/javaagent/src/test/java/SpringIntegrationAndRabbitTest.java

```diff
@@ -0,0 +1,174 @@
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_ADDRESS,
+                                s -> s.isIn("127.0.0.1", "0:0:0:0:0:0:0:1", null)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_PORT,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_TYPE, s -> s.isIn("ipv4", "ipv6", null)),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_SYSTEM,
+                                s -> s.isEqualTo("rabbitmq")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_DESTINATION_NAME,
+                                s -> s.isEqualTo("testTopic")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_OPERATION,
+                                s -> s.isEqualTo("publish")),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_MESSAGE_BODY_SIZE,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                MessagingIncubatingAttributes
+                                    .MESSAGING_RABBITMQ_DESTINATION_ROUTING_KEY,
+                                s -> s.isInstanceOf(String.class))),
+                // spring-cloud-stream-binder-rabbit listener puts all messages into a BlockingQueue
+                // immediately after receiving
+                // that's why the rabbitmq CONSUMER span will never have any child span (and
+                // propagate context, actually)
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("testTopic.anonymous.[-\\w]+ process"))
+                        .hasParent(trace.getSpan(6))
+                        .hasKind(SpanKind.CONSUMER)
+                        .hasAttributesSatisfyingExactly(
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_ADDRESS,
+                                s -> s.isIn("127.0.0.1", "0:0:0:0:0:0:0:1", null)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_PEER_PORT,
+                                l -> l.isInstanceOf(Long.class)),
+                            satisfies(
+                                NetworkAttributes.NETWORK_TYPE, s -> s.isIn("ipv4", "ipv6", null)),
+                            satisfies(
+                                MessagingIncubatingAttributes.MESSAGING_SYSTEM,
+                                s -> s.isEqualTo("rabbitmq")),
```

Comment:
maybe this is a nitpick, but for these `isEqualTo` assertions, couldn't you instead do

```suggestion
                            equalTo(MessagingIncubatingAttributes.MESSAGING_SYSTEM, "rabbitmq"),
```